redis/src/main/scala/zio/redis/Interpreter.scala

```diff
@@ -91,25 +99,194 @@ trait Interpreter {
 
     }
 
-    private[this] final class InMemory() extends Service {
-      def execute(command: Chunk[RespValue.BulkString]): zio.IO[RedisError, RespValue] =
+    private final class Test(
+      sets: TMap[String, Set[String]],
+      strings: TMap[String, String]
+    ) extends Service {
+      override def execute(command: Chunk[RespValue.BulkString]): zio.IO[RedisError, RespValue] =
         for {
           name   <- ZIO.fromOption(command.headOption).orElseFail(ProtocolError("Malformed command."))
           result <- runCommand(name.asString, command.tail).commit
         } yield result
 
       private[this] def runCommand(name: String, input: Chunk[RespValue.BulkString]): STM[RedisError, RespValue] =
         name match {
-          case api.Connection.Ping.name =>
+          case api.Connection.Ping.name  =>
             STM.succeedNow {
               if (input.isEmpty)
                 RespValue.bulkString("PONG")
               else
                 input.head
             }
+          case api.Sets.SAdd.name        =>
+            val key = input.head.asString
```

Comment:
I'd use `headOption` and handle the empty case.

Reply:
I think it doesn't make sense here.
The api defines that there has to be at least one argument. If one isn't present it really is a programming / unrecoverable error. The only alternative that makes sense here IMO is to die the fiber, but this will achieve that too.